experimental/javascript-wc-indexeddb/dist/components/todo-item/todo-item.component.js

```diff
@@ -0,0 +1,182 @@
+import template from "./todo-item.template.js";
+import { useDoubleClick } from "../../hooks/useDoubleClick.js";
+import { useKeyListener } from "../../hooks/useKeyListener.js";
+
+import globalStyles from "../../styles/global.constructable.js";
+import itemStyles from "../../styles/todo-item.constructable.js";
+
+class TodoItem extends HTMLElement {
+    static get observedAttributes() {
+        return ["itemid", "itemtitle", "itemcompleted"];
+    }
+
+    constructor() {
+        super();
+
+        // Renamed this.id to this.itemid and this.title to this.itemtitle.
+        // When the component assigns to this.id or this.title, this causes the browser's implementation of the existing setters to run, which convert these property sets into internal setAttribute calls. This can have surprising consequences.
+        // [Issue]: https://github.com/WebKit/Speedometer/issues/313
+        this.itemid = "";
+        this.itemtitle = "Todo Item";
+        this.itemcompleted = "false";
+        this.itemIndex = null;
+
+        const node = document.importNode(template.content, true);
+        this.item = node.querySelector(".todo-item");
+        this.toggleLabel = node.querySelector(".toggle-todo-label");
+        this.toggleInput = node.querySelector(".toggle-todo-input");
+        this.todoText = node.querySelector(".todo-item-text");
+        this.todoButton = node.querySelector(".remove-todo-button");
+        this.editLabel = node.querySelector(".edit-todo-label");
+        this.editInput = node.querySelector(".edit-todo-input");
+
+        this.shadow = this.attachShadow({ mode: "open" });
+        this.htmlDirection = document.dir || "ltr";
+        this.setAttribute("dir", this.htmlDirection);
+        this.shadow.adoptedStyleSheets = [globalStyles, itemStyles];
+        this.shadow.append(node);
+
+        this.keysListeners = [];
+
+        this.updateItem = this.updateItem.bind(this);
+        this.toggleItem = this.toggleItem.bind(this);
+        this.removeItem = this.removeItem.bind(this);
+        this.startEdit = this.startEdit.bind(this);
+        this.stopEdit = this.stopEdit.bind(this);
+        this.cancelEdit = this.cancelEdit.bind(this);
```

Comment:
this pattern of overriding the function with the bound function is a bit weird, I would prefer this happening to a separately named variable.

Reply:
ACK – let's keep it then.